src/main/java/de/tum/in/www1/artemis/web/websocket/RepositoryWebsocketService.java

```diff
@@ -0,0 +1,177 @@
+package de.tum.in.www1.artemis.web.websocket;
+
+import afu.org.checkerframework.checker.nullness.qual.Nullable;
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.ParticipationService;
+import de.tum.in.www1.artemis.service.UserService;
+import de.tum.in.www1.artemis.service.connectors.GitService;
+import de.tum.in.www1.artemis.web.rest.ParticipationResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.messaging.handler.annotation.DestinationVariable;
+import org.springframework.messaging.handler.annotation.MessageMapping;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.messaging.simp.SimpMessageSendingOperations;
+import org.springframework.stereotype.Controller;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.security.Principal;
+import java.util.Optional;
+
+@Entity
```

Comment:
do we really need the annotations @Entity and @Id here? These classes are not persisted in the database, so I guess we can remove them.
I would also suggest to put them into their own files as this is best practice in Java

Reply:
You are right about both - I made a commit to fix these issues.